services/auth-svc/domain/session/session.go

```diff
@@ -16,15 +16,17 @@ type Session struct {
 // New constructs a new Session with a new random access token
 // with the expiration based on expireAfter.
 func New(email mail.Address, expireAfter time.Duration) Session {
+	now := time.Now().UTC().Truncate(time.Second)
 	return Session{
 		AccessToken: NewAccessToken(),
 		Email:       email,
-		ExpireAt:    time.Now().UTC().Add(expireAfter),
+		ExpireAt:    now.Add(expireAfter.Truncate(time.Second)),
```

Comment:
Should truncate `now` and `expireAt` separately before adding them together.

Truncate `expireAt` in case caller accidentally pass `time.Duration` with a sub-second resolution.